data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/codec/JsonCodec.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.codec;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.model.record.Record;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * An implementation of {@link Codec} which parses JSON objects for arrays.
+ */
+public class JsonCodec implements Codec {
```

Comment:
@cmanning09 , That was part of the original S3 source proposal. I did change this recently for a couple reasons:

1. The structure of these logs is generally such that the individual record is simply part of a single array in the JSON.
2. Using JSON Pointer may not be possible with streaming presently. 

I believe that we could later add a JSON Pointer option that could be configured by users. If present, Data Prepper would find the list using JSON Pointer. But, this would result in having to load the entire JSON into memory.

Reply:
Thanks for clarifying.